src/celengine/solarsys.cpp

```diff
@@ -847,7 +861,16 @@ Body* CreateBody(const std::string& name,
     {
         auto oblateness = planetData->getNumber<float>("Oblateness");
         if (oblateness.has_value())
-            body->setSemiAxes(body->getRadius() * Eigen::Vector3f(1.0f, 1.0f - *oblateness, 1.0f));
+        {
+            if (*oblateness > 0.0f && *oblateness < 1.0f)
```

Comment:
Zero oblateness is a valid value corresponding to a sphere. If we disallow it, we prevent modifying an object from oblate to spherical.

Reply:
makes sense